### PR TITLE
Death parser

### PR DESCRIPTION
**Pull request by ewjax**
Parses for death conditions (You have been slain, or a tell to .death).

If 4 deaths occur in 120 seconds, with no signs of life from the player (casting, meleeing, or communicating), it will trigger a response.  The current response is to yell "Death Loop!" at the user and hope to gain their attention.  A more helpful response might be to kill the eqgame.exe process, but I understand that is currently ruled as out of bounds.

A series of unit tests is included, to exercise various portions of the code.

Outstanding tasks can be shown by searching for "todo".  There are currently three:
1. Make the 4 deaths and 120 second default values configurable in some manner.
2. Handle the sign-of-life communications edge case where tells show up as "{playername} ->", rather than "You tell".  I never could figure out how to get at the playername.
3. Strip away all user buff timers when death is detected

Comments welcome.  I made an attempt to match up coding styles, but that can be a highly personal thing, so if changes are needed in that regard, that's fine.

### CODE REVIEW

**Comment by ewjax** [comment]
Test comment, am I visible?

**Review by smasherprog — APPROVED** [review]

File: src/Services/Parsing/InvisParser.cs [hunk 1]
```diff
@@ -23,14 +26,24 @@ public bool Handle(string line, DateTime timestamp)
...
             }
             return false;
         }
 
         public InvisStatus? Parse(string line)
         {
-            return line == "You feel yourself starting to appear." ? InvisStatus.Fading : (InvisStatus?)null;
+            // this regex allows the parser to watch for the real phrase, but also to be tested by
+            // sending a tell while in-game to the non-existent user ".inv "
+            string pattern = @"(^\.inv )|(^You feel yourself starting to appear.)";
```
[the remark this answers is not in the document]
Comment (ewjax) [reply src/Services/Parsing/InvisParser.cs]: It is admittedly of marginal utility in this case, but it allows you to test the invis fading response in game by sending a tell to .inv.

[the remark this answers is not in the document]
Comment (ewjax) [reply src/Services/Parsing/InvisParser.cs]: The main goal was to incorporate regex matching, and the InvisParser class was a good starting point for that.  As mentioned, the ability to dynamically test this feature on the fly is of only marginal utility, but marginal > zero, so I left it.

File: src/Services/Parsing/DeathParser.cs [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+        }
+
+        //
+        // response if a deathloop condition is detected
+        // an ideal solution would be to kill the eqgame.exe process, however
+        // that is currently ruled out of bounds, so we will have to do something more benign for now
+        //
+        // returns the number of death timestamps currently in the tracking list
+        //
+        public int DeathLoopResponse()
```
[the remark this answers is not in the document]
Comment (ewjax) [reply src/Services/Parsing/DeathParser.cs]: I need to better understand the pub/sub pattern going on here.

**Pull request merged**